core/dbt/parser/manifest.py

```diff
@@ -1239,6 +1248,14 @@ def _process_refs_for_metric(manifest: Manifest, current_project: str, metric: M
                 should_warn_if_disabled=False,
             )
             continue
+        elif (
+            target_model.resource_type == NodeType.Model
+            and target_model.access == AccessType.Private
+        ):
+            # Metrics do not have a group and so can never reference private models
```

Comment:
There is no "group" field on a Metric node, but I see that there is on MetricConfig, so if it were set in config on a metric it would blow up in "update_parsed_node_config". Should I add a "group" field on Metric?

Reply:
Actually metrics doesn't even use update_parsed_node_config, so it wouldn't have blown up, but there wasn't a "group" field on metric. I added the group field and copied it from config to node level. I used the node level field in '_process_refs_..." calls.  If we're going to copy that field to node level we should probably use that consistently. Or not bother with the copying.